File: TransCoder_tokenized_test_set_functions/java/SUM_FAI_AJ_PAIRS_ARRAY_N_INTEGERS.java
public static int sum ( int a [ ], int n ) {
  Map < Integer, Integer > cnt = new HashMap < Integer, Integer > ( );
  int ans = 0, pre_sum = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    ans += ( i * a [ i ] ) - pre_sum;
    pre_sum += a [ i ];
    if ( cnt . containsKey ( a [ i ] - 1 ) ) ans -= cnt . get ( a [ i ] - 1 );
    if ( cnt . containsKey ( a [ i ] + 1 ) ) ans += cnt . get ( a [ i ] + 1 );
    if ( cnt . containsKey ( a [ i ] ) ) {
      cnt . put ( a [ i ], cnt . get ( a [ i ] ) + 1 );
    }
    else {
      cnt . put ( a [ i ], 1 );
    }
  }
  return ans;
}

File: results/java/SUM_FAI_AJ_PAIRS_ARRAY_N_INTEGERS.cpp
int f_filled ( int a [ ] , int n ) {
  map < int , int > cnt ;
  int ans = 0 , pre_f_filled = 0 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    ans += ( i * a [ i ] ) - pre_f_filled ;
    pre_f_filled += a [ i ] ;
    if ( cnt . find ( a [ i ] - 1 ) != cnt . end ( ) ) {
      ans -= cnt [ a [ i ] - 1 ] ;
    }
    if ( cnt . find ( a [ i ] + 1 ) != cnt . end ( ) ) {
      ans += cnt [ a [ i ] + 1 ] ;
    }
    if ( cnt . find ( a [ i ] ) != cnt . end ( ) ) cnt [ a [ i ] ] = cnt [ a [ i ] ] + 1 ;
    else cnt [ a [ i ] ] = 1 ;
  }
  return ans ;
}
